x/farming/keeper/proposal_handler.go

```diff
@@ -58,15 +58,18 @@ func (k Keeper) AddPublicPlanProposal(ctx sdk.Context, proposals []*types.AddReq
 				p.EpochAmount,
 			)
 
+			if err = msg.ValidateBasic(); err != nil {
```

Comment:
Do we need this validation? As far as I know this [Validate](https://github.com/tendermint/farming/blob/master/x/farming/types/proposal.go#L55) function is triggered when a user sends governance proposal to the network.